pkg/engine/artifacts.go

```diff
@@ -43,38 +43,38 @@ func kubernetesContainerAddonSettingsInit(profile *api.Properties) map[string]ku
 		DefaultAADPodIdentityAddonName: {
 			"kubernetesmasteraddons-aad-pod-identity-deployment.yaml",
 			"aad-pod-identity-deployment.yaml",
-			profile.OrchestratorProfile.KubernetesConfig.IsAADPodIdentityEnabled(),
+			profile.OrchestratorProfile.KubernetesConfig.IsAADPodIdentityEnabled() && !profile.IsAzureStackCloud(),
```

Comment:
@jadarsie you can actually move this one level up to `types.go` inside `IsAADPodIdentityEnabled()` which is unit tested (and add a UT case in types_test.go) if that makes sense.

Reply:
Sounds right, I would rather increase coverage.
Is it OK to change the signature of the `IsXXXEnabled` methods? 
It looks like I would need a reference to `Properties` or pass the boolean result of `IsAzureStackCloud`.

Reply:
yes that should be fine